bumble/device.py

```diff
@@ -868,6 +887,26 @@ async def disconnect(self, connection, reason):
             connection.remove_listener('disconnection_failure', pending_disconnection.set_exception)
             self.disconnecting = False
 
+    @asynccontextmanager
```

Comment:
I like the idea of a context manager for connections. Maybe we should have the `Connection` object itself implement the context management protocol, so that you would simply use: `with await device.connect(xyz) as connection`. (similar to how the File object implements the context management protocol, so you can choose to call `open(xyz)` or `with open(xyz) as input`).
This would also work regardless of whether you're intending to use GATT or not (Peer is mostly an abstraction for the GATT client functions, not something that's currently useful for Classic connections, and `auto_discovery` also only applies to GATT, not to all connections).
The `auto_discovery` functionality could then be moved to the `Peer` constructor, something like `peer = Peer(connection, auto_discovery=True)`.

Reply:
Adding it to the `Connection` object directly makes a lot of sense. While `Peer` is only applicable to GATT, it is the more common use-case in the examples. I think there is still value in a context manager that can do the initial setup. For GATT clients, returning a `Peer` and performing discovery is a common flow.

Would you be amenable to adding the context manager to `Connection`, add a context manager to `Peer` for performing discovery and adding a `client=None` to the `.connect` method that takes a class that can be instantiated with a connection and is a context manager. So you can do `async with device.connect(addr, client=Peer) as peer` or `async with device.connect(addr) as conn`.

Reply:
That would work. But we'd lose the `auto_discovery` argument, which is useful. Maybe we can offer the simple `device.connect(addr)` that returns a `Connection` for the generic use case (whether or not the caller then decides to instantiate a `Peer`), plus a convenience `device.connect_for_gatt(addr, auto_discovery=False)` that would return a context-managed `Peer` with optional auto-discovery (like you have now)